core/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -498,9 +498,45 @@ trait VacuumCommandImpl extends DeltaCommand {
       basePath: Path,
       relativizeIgnoreError: Boolean,
       isBloomFiltered: Boolean): Seq[String] = {
-    getActionRelativePath(action, fs, basePath, relativizeIgnoreError).map { relativePath =>
+    val paths = getActionRelativePath(action, fs, basePath, relativizeIgnoreError).map {
+      relativePath =>
       Seq(relativePath) ++ getAllSubdirs("/", relativePath, fs)
     }.getOrElse(Seq.empty)
+
+    val deletionVectorPath =
+      getDeletionVectorRelativePath(action, fs, basePath, relativizeIgnoreError).map(pathToString)
+
+    paths ++ deletionVectorPath.toSeq
+  }
+
+    /** Returns the relative path of the on-disk deletion vector. */
+  protected def getDeletionVectorRelativePath(
+      action: FileAction,
+      fs: FileSystem,
+      basePath: Path,
+      relativizeIgnoreError: Boolean
+  ): Option[Path] = {
+    val dv = action match {
+      case a: AddFile if a.deletionVector != null =>
+        Some(a.deletionVector)
+      case r: RemoveFile if r.deletionVector != null =>
+        Some(r.deletionVector)
+      case _ => None
+    }
+
+    dv match {
+      case Some(dv) if dv.isOnDisk =>
+        if (dv.isRelative) {
+          Some(dv.absolutePath(new Path(".")))
+        } else {
+          Some(DeltaFileOperations.tryRelativizePath(
+            fs,
+            basePath,
+            dv.absolutePath(basePath),
+            relativizeIgnoreError))
```

Comment:
I don't think this makes sense for DVs. The Delta issue that introduced potentially absolute paths that refer to a local file never existed for DVs, so this will never have any effect. You can just as well simply return `dv.absolutePath(basePath)`. Or even `None`, since VACUUM wouldn't touch paths in another directory anyway.